lib/BoxdropHelper.class.php

```diff
@@ -45,10 +47,15 @@ public static function getAdminDirPath()
 					$absolute_element = $root_dir.$dir_content;
 					if (is_dir($absolute_element))
 					{
-						if (strpos($dir_content, 'admin') !== false)
+						$index_file = $absolute_element.'/index.php';
+						if (file_exists($index_file))
 						{
-							$admin_dir = $absolute_element;
-							break;
+							$file_contents = file_get_contents($index_file);
+							if (strpos($file_contents, "define('_PS_ADMIN_DIR_', getcwd()") !== false)  
+							{								
+								$admin_dir = $absolute_element;
+								break;
+							}
```

Comment:
I guess we could improve the way you create the admin environment for the next version of BoxDrop, in example with ModuleAdminController (and a ModuleFrontController for the front-office). When the controller will be called, you will be sure to be in an admin context, and all your checks won't be needed anymore. Please have a look on it, I'm sure your code will be safer and cleaner.